dss/storage/checkout/bundle.py

```diff
@@ -72,50 +72,47 @@ def verify_checkout(
         replica: Replica,
         bundle_uuid: str,
         bundle_version: typing.Optional[str],
-        token: typing.Optional[str]
 ) -> typing.Tuple[str, bool]:
     """
     Ensures that for a specified bundle either:
     a) a checkout exists
     b) a job processing a bundle checkout exists
 
     This function will first check if a valid checkout exists. Else, it will
-    create or use a supplied checkout job token to process a bundle checkout job.
+    create or use a checkout job token to process a bundle checkout job.
 
     :param replica: Cloud replica
     :param bundle_uuid: Bundle UUID
     :param bundle_version: Bundle version
-    :param token: Token used to track the status of a checkout job.
-    :return: (checkout_job_token, is_checkout_ready)
+    :return: is_checkout_ready
     """
     if _is_checkout_valid(replica, bundle_uuid, bundle_version):
         if _is_checkout_stale(replica, bundle_uuid, bundle_version):
             start_bundle_checkout(replica, bundle_uuid, bundle_version, dst_bucket=replica.checkout_bucket)
         return "", True
 
     decoded_token: dict
+
+    async_key = f"{replica.name}/{bundle_uuid}.{bundle_version}"
```

Comment:
It seems the composition of the `async_key` would need to include the bucket name to avoid state getting conflated in the following cases of concurrent checkouts of the same bundle/version:
* Concurrent checkout to DSS checkout bucket and to a user-provided bucket
* Concurrent checkouts to two different user-provided buckets